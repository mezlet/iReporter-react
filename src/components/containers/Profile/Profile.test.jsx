/* eslint-disable no-unused-vars */
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import connectProfile, { Profile } from './Profile';

const props = {
  getProfile: jest.fn(),
  profile: {
    data: {
      resolved: '',
      pending: '',
      posts: '',
      rejected: '',
      user: {},
      userIncidents: {}
    }
  }
};

const initialState = {
  isLoading: false,
  success: false,
  incident: {},
  errors: {},
  message: ''
};

const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);

const connectedWrapper = shallow(
  <Provider store={store}>
    <Router>
      <connectProfile {...props} />
    </Router>
  </Provider>
);

const wrapper = shallow(<Profile {...props} />);

describe('<Profile/> rendering', () => {
  it('should render without crashing', () => {
    expect(connectedWrapper).toMatchSnapshot();
  });
});
