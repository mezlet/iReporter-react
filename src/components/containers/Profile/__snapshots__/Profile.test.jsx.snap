// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> rendering should render without crashing 1`] = `ShallowWrapper {}`;
